blocks/bricks/sequence_generators.py

```diff
@@ -290,38 +290,80 @@ def readout(self, **kwargs):
         pass
 
 
-@add_metaclass(ABCMeta)
-class Readout(AbstractReadout):
+class Readout(AbstractReadout, Initializable):
     """Readout brick with separated emitting and feedback parts.
 
     Parameters
     ----------
+    source_names : list
+        A list of the source names (outputs) that are needed for the
+        readout part e.g. ``['states']`` or ``['states', 'glimpses']``.
```

Comment:
So what about this? There is currently no glimpse named `glimpses`, only `weighted_averages`.